lib/utils/isPromiseLike.ts

```diff
@@ -0,0 +1,3 @@
+export function isPromise(data: any): data is Promise<unknown> {
+  return data && data.then && data.catch;
```

Comment:
In JS a promise-like is considered any object that has the `then` method. You need to check that there is `then` and that it's a function.